src/plugins/embeddable/public/plugin.tsx

```diff
@@ -76,18 +106,47 @@ export interface EmbeddableSetup {
     id: string,
     factory: EmbeddableFactoryDefinition<I, O, E>
   ) => () => EmbeddableFactory<I, O, E>;
+  /**
+   * @deprecated
+   */
   registerEnhancement: (enhancement: EnhancementRegistryDefinition) => void;
+  /**
+   * @deprecated
+   */
   setCustomEmbeddableFactoryProvider: (customProvider: EmbeddableFactoryProvider) => void;
 }
 
 export interface EmbeddableStart extends PersistableStateService<EmbeddableStateWithType> {
+  /**
+   * Registers an async {@link ReactEmbeddableFactory} getter.
+   */
+  registerReactEmbeddableFactory: <
```

Comment:
Why is registerReactEmbeddableFactory getting added to both Setup and Start API? I think this is confusing and goes against the principle of only have in a single way to do things. The registry should only exist on either Setup or Start but not both.

Reply:
It's on both to match the legacy embeddable factory. Likely it should be moved to the `setup` phase only, as I think that's a better time to register embeddables, but I made it this way with intentions to limit it more later. 

I can look into changing it in this PR to exist only in setup.

Reply:
Resolved this in [c1f7196](https://github.com/elastic/kibana/pull/182621/commits/c1f7196200a3d5bf4763969d90cc9ba202b1091a)